seatunnel-connectors-v2/connector-cdc/connector-cdc-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mongodb/source/fetch/MongodbScanFetchTask.java

```diff
@@ -0,0 +1,247 @@
+
+                SourceRecord snapshotRecord =
+                        buildSourceRecord(sourceConfig, collectionId, keyDocument, valueDocument);
+
+                changeEventQueue.enqueue(new DataChangeEvent(snapshotRecord));
+            }
+
+            ChangeStreamOffset highWatermark = dialect.displayCurrentOffset(sourceConfig);
+            log.info(
+                    "Snapshot step 3 - Determining high watermark {} for split {}",
+                    highWatermark,
+                    snapshotSplit);
+            changeEventQueue.enqueue(
+                    new DataChangeEvent(
+                            WatermarkEvent.create(
+                                    createWatermarkPartitionMap(collectionId.identifier()),
+                                    "__mongodb_watermarks",
+                                    snapshotSplit.splitId(),
+                                    WatermarkKind.HIGH,
+                                    highWatermark)));
+
+            log.info(
+                    "Snapshot step 4 - Back fill stream split for snapshot split {}",
+                    snapshotSplit);
+            final IncrementalSplit dataBackfillSplit =
+                    createBackfillStreamSplit(lowWatermark, highWatermark);
+            final boolean streamBackfillRequired =
+                    dataBackfillSplit.getStopOffset().isAfter(dataBackfillSplit.getStartupOffset());
+
+            if (!streamBackfillRequired) {
+                changeEventQueue.enqueue(
+                        new DataChangeEvent(
+                                WatermarkEvent.create(
+                                        createWatermarkPartitionMap(collectionId.identifier()),
+                                        "__mongodb_watermarks",
+                                        dataBackfillSplit.splitId(),
+                                        WatermarkKind.END,
+                                        dataBackfillSplit.getStopOffset())));
+            } else {
+                MongodbStreamFetchTask dataBackfillTask =
+                        new MongodbStreamFetchTask(dataBackfillSplit);
+                dataBackfillTask.execute(taskContext);
+            }
+        } catch (Exception e) {
+            throw new MongodbConnectorException(
+                    ILLEGAL_ARGUMENT,
+                    String.format(
+                            "Execute snapshot read subtask for mongodb split %s fail",
+                            snapshotSplit));
+        } finally {
+            taskRunning = false;
+        }
+    }
+
+    @NotNull private MongoCursor<RawBsonDocument> getSnapshotCursor(
+            @NotNull SnapshotSplit snapshotSplit, MongodbSourceConfig sourceConfig) {
+        MongoClient mongoClient = createMongoClient(sourceConfig);
+        MongoCollection<RawBsonDocument> collection =
+                getMongoCollection(mongoClient, snapshotSplit.getTableId(), RawBsonDocument.class);
```

Comment:
Please add debug log to let us know which table split is being read .

`LOG.debug("Snapshotting table {}", tableId);`